src/components/sbb-icon/sbb-icon.tsx

```diff
@@ -0,0 +1,185 @@
+   * that start with `on`, such as `onclick`.
+   * @default true
+   */
+  @Prop() public sanitize = true;
+
+  public connectedCallback(): void {
+    this._loadSvgIcon(this.name);
+  }
+
+  /**
+   * Update the icon svg content on name change
+   */
+  @Watch('name')
+  public updateSvgIcon(newValue: string, oldValue: string): void {
+    if (newValue !== oldValue) {
+      this._element.removeAttribute('aria-label');
+      this._loadSvgIcon(newValue);
+    }
+  }
+
+  private _loadSvgIcon(iconName: string): void {
+    if (!iconName) {
+      return;
+    }
+
+    const [namespace, name] = this._splitIconName(iconName);
+
+    if (namespace) {
+      this._svgNamespace = namespace;
+    }
+
+    if (name) {
+      this._svgName = name;
+    }
+
+    this._ariaHidden = this._getAriaHidden();
+
+    // generate a default label in case user does not provide their own
+    if (this._ariaHidden === 'false' && this._svgName) {
+      this._ariaLabel = this._getAriaLabel() || `Icon ${this._svgName.replace(/-/g, ' ')}`;
+    }
+
+    // try to load SVG from registered icons
+    this._svgIcon = this._loadRegisteredIcon();
+
+    if (this._svgIcon) {
+      return;
+    }
+
+    const url = this._resolveSvgUrl();
+
+    if (url) {
+      if (cachedIcons.has(url)) {
+        // if it's already loaded
+        this._svgIcon = cachedIcons.get(url);
+      } else {
+        getSvgContent(url, this.sanitize).then(() => (this._svgIcon = cachedIcons.get(url)));
+      }
+    }
+  }
+
+  private _splitIconName(iconName: string): [string, string] {
+    if (!iconName) {
+      return ['', ''];
+    }
+    const parts = iconName.split(':');
+    switch (parts.length) {
+      case 1:
+        return ['', parts[0]]; // Use default namespace
+      case 2:
+        return parts as [string, string];
+      default:
+        throw Error(`Invalid icon name: "${iconName}"`);
+    }
+  }
+
+  private _loadRegisteredIcon(): string {
+    return registeredIcons.get(this.name);
+  }
+
+  private _resolveSvgUrl(): string {
+    if (iconNamespaces.has(this._svgNamespace)) {
+      return `${iconNamespaces.get(this._svgNamespace)}${this._svgName}.svg`;
+    } else {
+      throw Error(
+        `Unable to find the namespace "${this._svgNamespace}". Please register your custom namespace through the icon registry API.`
+      );
+    }
+  }
+
+  private _getAriaHidden(): string {
+    return this._element.getAttribute('aria-hidden');
+  }
+
+  private _getAriaLabel(): string {
+    return this._element.getAttribute('aria-label');
+  }
+
+  public render(): JSX.Element {
+    return (
+      <Host
+        aria-hidden={this._ariaHidden === 'false' ? 'false' : 'true'}
+        aria-label={
+          this._ariaLabel !== undefined && this._ariaHidden === 'false' ? this._ariaLabel : null
+        }
+        role="img"
+        class={`sbb-icon ${this._svgName || ''}`}
+      >
+        {this._svgIcon ? (
+          <div class="sbb-icon-inner" innerHTML={this._svgIcon}></div>
```

Comment:
Can the svg content not be attached to the host directly?

Reply:
Sure